cogs/Dictionary.py

```diff
@@ -0,0 +1,506 @@
+                elif audio_element.has_attr("data-src-ogg"):
+                    pronunciation.audio_url = str(audio_element["data-src-ogg"])
+
+            return pronunciation
+
+        return None
+
+
+@dataclass
+class Example:
+    sentence: str
+    labels: str | None  # i.e. "(informal) (ironic)"
+
+    def __str__(self) -> str:
+        """
+        Formats the example for Discord markdown.
+
+        :return: a string like "*(ironic)* Oh great, they left without us."
+        """
+
+        if self.labels:
+            return f"*{self.labels}* {self.sentence}"
+
+        return self.sentence
+
+
+@dataclass
+class Sense:
+    definition: str
+    grammar: str | None  # i.e. "[usually before noun]"
+    labels: str | None  # i.e. "(informal) (ironic)"
+    examples: list[Example]
+
+    def __str__(self) -> str:
+        """
+        Formats the sense for Discord markdown without any examples.
+
+        :return: a string like "*[only before noun] (informal)* used to emphasize an adjective of size or quality"
+        """
+        if self.grammar and self.labels:
+            return f"*{self.grammar} {self.labels}* {self.definition}"
+        elif self.grammar:
+            return f"*{self.grammar}* {self.definition}"
+        elif self.labels:
+            return f"*{self.labels}* {self.definition}"
+
+        return self.definition
+
+
+@dataclass
+class SimilarWord:
+    word: str
+    url: str
+    part_of_speech: str | None
+
+    async def fetch(self) -> "WordInformation":
+        """
+        Fetches the word information based on the URL.
+
+        :return: a WordInformation
+        """
+        status_code, soup = await fetch_page_cached(self.url)
+        assert (
+            status_code == 200
+        ), "Got a non-200 response while fetching a URL that was provided directly from Oxford."
+        return parse_oxford_definition_page(self.url, soup)
+
+
+@dataclass
+class WordInformation:
+    US_FLAG_EMOJI = (
+        "\N{REGIONAL INDICATOR SYMBOL LETTER U}\N{REGIONAL INDICATOR SYMBOL LETTER S}"
+    )
+    GB_FLAG_EMOJI = (
+        "\N{REGIONAL INDICATOR SYMBOL LETTER G}\N{REGIONAL INDICATOR SYMBOL LETTER B}"
+    )
+    SPEAKING_HEAD_EMOJI = "\N{SPEAKING HEAD IN SILHOUETTE}\N{VARIATION SELECTOR-16}"
+
+    url: str
+    word: str
+    part_of_speech: str | None
+    us_pronunciation: Pronunciation | None
+    gb_pronunciation: Pronunciation | None
+    senses: list[Sense]
+    similar_words: list[
+        SimilarWord
+    ]  # may include words with identical spellings, but differing parts of speech
+
+    def build_embed(self) -> discord.Embed:
+        """
+        Builds an embed that contains all the relevant information.
+        Will only include the first 5 definitions with max 3 examples each, or 5 examples if there is only one definition.
+
+        :return: a discord.Embed
+        """
+        title = f"*{self.word.title()}*"
+        footer = f"Retrieved from Oxford Learner's Dictionary by Homework Help"
```

Comment:
Not totally sure, but I think that this is legally required.

Reply:
Probably best to do that, better safe than sorry